### PR TITLE
feature #773: Created endpoint for updating cathedras

### LINKED ISSUE #773: Create endpoint for updating cathedras
Your task is to create an endpoint for updating cathedra by id. Also create an update body with optional fields. 

The updated cathedra should be in the response.

### PR DESCRIPTION
**Pull request by JessFreak**
- created CathedraMapper
- created DbCathedra
- created update methods in Cathedra Repository and Service
- created UpdateCathedraDTO
- changed and added Swagger documentation

closes #773

### CODE REVIEW

File: src/v2/api/dtos/CreateCathedraDTO.ts [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
+  @IsNotEmpty(validationOptionsMsg('Cathedra name can not be empty'))
   @Matches(
     createRegex(UKR_REGEX, UKRSPEC_REGEX),
     validationOptionsMsg('Cathedra name is incorrect (A-Я(укр.)\\-\' )'),
   )
     name: string;
 
-  @ApiProperty()
+  @ApiProperty({
+    description: 'The name of the cathedra',
```
Comment (illiakroshka) [inline src/v2/api/dtos/CreateCathedraDTO.ts:19]: Maybe here better to write "The abbreviation of the cathedra"

**Review by illiakroshka — CHANGES_REQUESTED** [review]

**Review by illiakroshka — APPROVED** [review]

**Review by ArtemLyah — APPROVED** [review]

File: src/v2/modules/CathedraModule.ts [hunk 2]
```diff
@@ -2,11 +2,13 @@ import { Module } from '@nestjs/common';
...
+import { TeacherModule } from './TeacherModule';
+import { CathedraMapper } from '../mappers/CathedraMapper';
 
 @Module({
   controllers: [CathedraController],
-  providers: [CathedraService],
-  exports: [CathedraService],
-  imports: [AccessModule],
+  providers: [CathedraService, CathedraMapper],
+  exports: [CathedraService, CathedraMapper],
```
Comment (fokaaas) [inline src/v2/modules/CathedraModule.ts:11]: You have to add it to mapper module

File: src/v2/modules/TeacherModule.ts [hunk 3]
```diff
@@ -11,11 +11,12 @@ import { ConfigurationModule } from './ConfigModule';
...
 import { MapperModule } from './MapperModule';
 import { CommentsQueryPipe } from '../api/pipes/CommentsQueryPipe';
+import { TeacherMapper } from '../mappers/TeacherMapper';
 
 @Module({
   controllers: [TeacherController, DisciplineTeacherController],
-  providers: [TeacherService, DisciplineTeacherService, TelegramAPI, SubjectByIdPipe, CommentsQueryPipe],
-  exports: [TeacherService, DisciplineTeacherService],
+  providers: [TeacherService, DisciplineTeacherService, TelegramAPI, SubjectByIdPipe, CommentsQueryPipe, TeacherMapper],
+  exports: [TeacherService, DisciplineTeacherService, TeacherMapper],
```
Comment (fokaaas) [inline src/v2/modules/TeacherModule.ts:19]: You have to add it to mapper module

File: src/v2/api/dtos/CreateCathedraDTO.ts [hunk 4]
```diff
@@ -1,19 +1,23 @@
...
 
 export class CreateCathedraDTO {
-  @ApiProperty()
+  @ApiProperty({
+    description: 'The name of the cathedra',
+  })
   @MinLength(5, validationOptionsMsg('Cathedra name is too short (min: 5)'))
   @MaxLength(40, validationOptionsMsg('Cathedra name is too long (max: 40)'))
-  @IsOptional()
+  @IsNotEmpty(validationOptionsMsg('Cathedra name can not be empty'))
```
Comment (fokaaas) [inline src/v2/api/dtos/CreateCathedraDTO.ts:11]: It's optional

**Review by fokaaas — CHANGES_REQUESTED** [review]

**Review by fokaaas — APPROVED** [review]

**Pull request merged**